lib/fizzy/leb128.hpp

```diff
@@ -46,13 +52,29 @@ std::pair<T, const uint8_t*> leb128s_decode(const uint8_t* input, const uint8_t*
         result |= static_cast<T_unsigned>((static_cast<T_unsigned>(*input) & 0x7F) << result_shift);
         if ((*input & 0x80) == 0)
         {
-            // sign extend
-            if ((*input & 0x40) != 0 && (result_shift + 7 < sizeof(T_unsigned) * 8))
+            if (result_shift + 7 < sizeof(T_unsigned) * 8)
             {
-                constexpr auto all_ones = std::numeric_limits<T_unsigned>::max();
-                const auto mask = static_cast<T_unsigned>(all_ones << (result_shift + 7));
-                result |= mask;
+                // non-terminal byte of the encoding, extend the sign bit, if needed
+                if ((*input & 0x40) != 0)
+                {
+                    constexpr auto all_ones = std::numeric_limits<T_unsigned>::max();
+                    const auto mask = static_cast<T_unsigned>(all_ones << (result_shift + 7));
+                    result |= mask;
+                }
             }
+            else
+            {
+                // terminal byte of the encoding, need to check unused bits
+                const auto expected =
+                    (static_cast<uint8_t>(static_cast<T>(result) >> result_shift) & 0x7f);
```

Comment:
Nitpick, in the rest of file you use `0x7F` (uppercase).

Reply:
fixed